src/Html.php

```diff
@@ -607,83 +549,88 @@ public static function resetInput($label = 'Reset', $options = []): string
-     * If a value is null, the corresponding attribute will not be rendered.
+     * @param array $options the tag options in terms of name-value pairs. These will be rendered as the attributes of
+     * the resulting tag. The values will be HTML-encoded using {@see encode()}. If a value is null, the corresponding
+     * attribute will not be rendered.
      * See {@see renderTagAttributes()} for details on how attributes are being rendered.
      * @return string the generated password input tag
      */
-    public static function passwordInput($name, $value = null, $options = []): string
+    public static function passwordInput(string $name, string $value = null, array $options = []): string
     {
         return static::input('password', $name, $value, $options);
     }
 
     /**
      * Generates a file input field.
-     * To use a file input field, you should set the enclosing form's "enctype" attribute to
-     * be "multipart/form-data". After the form is submitted, the uploaded file information
-     * can be obtained via $_FILES[$name] (see PHP documentation).
+     *
+     * To use a file input field, you should set the enclosing form's "enctype" attribute to be "multipart/form-data".
+     * After the form is submitted, the uploaded file information can be obtained via $_FILES[$name]
+     * (see PHP documentation).
      * @param string $name the name attribute.
      * @param string $value the value attribute. If it is null, the value attribute will not be generated.
      * @param array $options the tag options in terms of name-value pairs. These will be rendered as
-     * the attributes of the resulting tag. The values will be HTML-encoded using {@see encode()}.
-     * If a value is null, the corresponding attribute will not be rendered.
+     * the attributes of the resulting tag. The values will be HTML-encoded using {@see encode()}. If a value is null,
+     * the corresponding attribute will not be rendered.
      * See {@see renderTagAttributes()} for details on how attributes are being rendered.
      * @return string the generated file input tag
      */
-    public static function fileInput($name, $value = null, $options = []): string
+    public static function fileInput(string $name, string $value = null, array $options = []): string
     {
         return static::input('file', $name, $value, $options);
     }
 
     /**
      * Generates a text area input.
+     *
      * @param string $name the input name
-     * @param string $value the input value. Note that it will be encoded using {@see encode()}.
-     * @param array $options the tag options in terms of name-value pairs. These will be rendered as
-     * the attributes of the resulting tag. The values will be HTML-encoded using {@see encode()}.
-     * If a value is null, the corresponding attribute will not be rendered.
+     * @param string|null $value the input value. Note that it will be encoded using {@see encode()}.
```

Comment:
Is it allowing null?

Reply:
I have not wanted to touch many functions, only the necessary default comes like this, if you want we do not allow null, and you have to modify the test that verifies if it is null can we place ?string

Reply:
IMO `null` should be supported in all input methods - it is pretty common to have `null` as a default value in empty models.